src/analysis/modals/RuntimeErrorModal.test.tsx

```diff
@@ -0,0 +1,129 @@
+import { act, screen } from '@testing-library/react';
+import React from 'react';
+import { RuntimeBasics } from 'src/libs/ajax/leonardo/providers/LeoRuntimeProvider';
+import { asMockedFn, mockNotifications, renderWithAppContexts as render } from 'src/testing/test-utils';
+
+import { RuntimeErrorModal, RuntimeErrorProvider, text } from './RuntimeErrorModal';
+
+describe('RuntimeErrorModal', () => {
+  it('renders basic error', async () => {
+    // Arrange
+    const runtime: RuntimeBasics = {
+      cloudContext: {
+        cloudProvider: 'GCP',
+        cloudResource: 'myCloudResource',
+      },
+      runtimeName: 'myRuntime',
+      googleProject: 'myProject',
+      workspaceId: 'myWorkspace',
+    };
+    const errorProvider: RuntimeErrorProvider = {
+      errorInfo: jest.fn(),
+    };
+    asMockedFn(errorProvider.errorInfo).mockResolvedValue({
+      errorType: 'ErrorList',
+      errors: [{ errorMessage: 'Runtime went BOOM!', errorCode: 123, timestamp: '0' }],
+    });
+    const onDismiss = jest.fn();
+
+    // Act
+    await act(() =>
+      render(<RuntimeErrorModal runtime={runtime} onDismiss={onDismiss} errorProvider={errorProvider} />)
+    );
```

Comment:
Suggest extracting code repeated in several tests into a `setup` function.

https://medium.com/@kolodny/testing-with-sifers-c9d6bb5b362

Reply:
yeah.... but they are all just a little different from eachother.

Reply:
The `setup` function can take options for test-specific configuration.